contracts/extensions/ERC404UniswapV3Exempt.sol

```diff
@@ -0,0 +1,82 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {ERC404} from "../ERC404.sol";
+import {IPeripheryImmutableState} from "@uniswap/v3-periphery/contracts/interfaces/IPeripheryImmutableState.sol";
+
+abstract contract ERC404UniswapV3Exempt is ERC404 {
+  error ERC404UniswapV3ExemptFactoryMismatch();
+  error ERC404UniswapV3ExemptWETH9Mismatch();
+
+  constructor(address uniswapV3Router_, address uniswapV3NonfungiblePositionManager_) {
+    IPeripheryImmutableState uniswapV3Router = IPeripheryImmutableState(
+      uniswapV3Router_
+    );
+
+    // Set the Uniswap v3 swap router as exempt.
+    _setERC721TransferExempt(uniswapV3Router_, true);
+
+    IPeripheryImmutableState uniswapV3NonfungiblePositionManager = IPeripheryImmutableState(
+      uniswapV3NonfungiblePositionManager_
+    );
+
+    // Set the Uniswap v3 nonfungible position manager as exempt.
+    _setERC721TransferExempt(uniswapV3NonfungiblePositionManager_, true);
```

Comment:
Thoughts on overriding transfer exempt function and checking immutable variables? Not necessary, but would save gas, though this seems like the safer approach.